surch/surch.py

```diff
@@ -1,165 +1,349 @@
+        :type repo_kind: basestring
+        :param url_kind: url kind (git_url, ssh_url, clone_url, svn_url) default:'clone_url'
+        :type url_kind: basestring
+        :param repo_per_page: this for getting the MAX information in 1 page default:100
+        :type repo_per_page: int
+        :return:
+        """
         github_repo_urls = []
-        data = requests.get(GET_GIT_REPO_DETAILS_API_URL.format(organization, repo_per_page), auth=(user, password))
-        data = data.json()
-        for repo_data in data:
-            github_repo_urls.append(repo_data['clone_url'])
+        data = requests.get(GET_ORG_DETAILS.format(organization), auth=(git_user, git_password))
+
+        repo_num = data.json()['{0}_repos'.format(repo_kind)]
+        last_page_num = repo_num / repo_per_page
+        if (repo_num % repo_per_page) > 0:
+            # Adding 2 because 1 for the extra repos that mean more page, and 1 for the next for loop.
+            last_page_num += 2
+
+        for page_num in range(1, last_page_num):
+            data = requests.get(GET_GIT_REPO_DETAILS_API_URL.format(organization, repo_kind, repo_per_page, page_num),
+                                auth=(git_user, git_password))
+            data = data.json()
+            for repo_data in data:
+                github_repo_urls.append(repo_data[url_kind])
         return github_repo_urls
 
-    def clone_repo(self, github_repo_urls, path):
-        ''' clone or pull git repos'''
+    def clone_repo(self, github_repo_urls, local_path, organization):
+        """
+        This method run clone or pull for the repo list
+        :param github_repo_urls: list of repository git url
+        :type github_repo_urls: (tupe, list)
+        :param local_path: this path contain the repos clone
+        :type local_path: basestring
+        :param organization: organization name
+        :type organization: basestring
+        :return: cloned repo
+        """
         start = time.time()
         for url in github_repo_urls:
-            name_git = url.rsplit('/', 1)[-1]
-            full_path = os.path.join(path, name_git.rsplit('.', 1)[0])
-            if os.path.exists(full_path):
-                print 'pull request (update) {0} repo. from {1} org/user.'.format(name_git, self.organization)
-                git_pull = subprocess.check_output(GIT_PULL_COMMAND.format(full_path), shell=True)
-            else:
-                print 'clone {0} repo. from {1} org/user.'.format(name_git, self.organization)
-                git_clone = subprocess.check_output(GIT_CLONE_COMMAND.format(str(url), full_path), shell=True)
-        print 'git_clone\pull time: ' + print_performance(start, time.time()) + ' seconds'
-
-    def get_skipped_repo(self, ignore_list, path):
-        ''' get repo list to ignore and create full path list to ignore '''
+            repository_name = (url.rsplit('/', 1)[-1]).rsplit('.', 1)[0]
```

Comment:
instead of parsing the repo url, build a list of dicts containing repo data which can be used across surch